File: lib/src/notification/screen_capture_service_info.dart
import 'package:equatable/equatable.dart';
import 'package:stringee_plugin/stringee_plugin.dart';

class ScreenCaptureServiceInfo extends Equatable {
  // Channel info
  final String channelId;
  final String channelName;
  final String channelDescription;
  final LockScreenVisibility lockscreenVisibility;

  // Notification info
  final int id;
  final NotificationIcon? icon;
  final String? contentTitle;
  final String? contentText;

  // Other information
  final String clientId;
  final String callId;

  ScreenCaptureServiceInfo({
    required this.channelId,
    required this.channelName,
    required this.channelDescription,
    this.lockscreenVisibility = LockScreenVisibility.Private,
    required this.id,
    this.icon,
    this.contentTitle,
    this.contentText,
    required this.clientId,
    required this.callId,
  });

  Map<String, dynamic> toJson() => {
        'channelId': channelId.trim(),
        'channelName': channelName.trim(),
        'channelDescription': channelDescription.trim(),
        'lockscreenVisibility': lockscreenVisibility.value,
        'id': id,
        'uuid': clientId.trim(),
        'callId': callId.trim(),
        if (icon != null) 'icon': icon?.toJson(),
        if (contentTitle != null) 'contentTitle': contentTitle?.trim(),
        if (contentText != null) 'contentText': contentText?.trim(),
      };

  @override
  List<Object?> get props => [
        channelId,
        channelName,
        channelDescription,
        lockscreenVisibility,
        id,
        icon,
        contentTitle,
        contentText,
        clientId,
        callId,
      ];
}

File: android/src/main/java/com/stringee/stringeeflutterplugin/notification/model/ScreenCaptureServiceInfo.java
package com.stringee.stringeeflutterplugin.notification.model;

import android.app.Notification;

public class ScreenCaptureServiceInfo {
    // Channel info
    private String channelId;
    private String channelName;
    private String channelDescription;
    private int lockscreenVisibility = Notification.VISIBILITY_PUBLIC;

    // Notification info
    private int id;
    private String iconSource;
    private int sourceFrom;
    private String contentTitle;
    private String contentText;

    // Other information for call
    private String clientId;
    private String callId;

    public String getChannelId() {
        return channelId;
    }

    public void setChannelId(String channelId) {
        this.channelId = channelId;
    }

    public String getChannelName() {
        return channelName;
    }

    public void setChannelName(String channelName) {
        this.channelName = channelName;
    }

    public String getChannelDescription() {
        return channelDescription;
    }

    public void setChannelDescription(String channelDescription) {
        this.channelDescription = channelDescription;
    }

    public int getLockscreenVisibility() {
        switch (lockscreenVisibility) {
            case -1:
                return Notification.VISIBILITY_SECRET;
            case 1:
                return Notification.VISIBILITY_PUBLIC;
            default:
                return Notification.VISIBILITY_PRIVATE;
        }
    }

    public void setLockscreenVisibility(int lockscreenVisibility) {
        this.lockscreenVisibility = lockscreenVisibility;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getIconSource() {
        return iconSource;
    }

    public void setIconSource(String iconSource) {
        this.iconSource = iconSource;
    }

    public int getSourceFrom() {
        return sourceFrom;
    }

    public void setSourceFrom(int sourceFrom) {
        this.sourceFrom = sourceFrom;
    }

    public String getContentTitle() {
        return contentTitle;
    }

    public void setContentTitle(String contentTitle) {
        this.contentTitle = contentTitle;
    }

    public String getContentText() {
        return contentText;
    }

    public void setContentText(String contentText) {
        this.contentText = contentText;
    }

    public String getClientId() {
        return clientId;
    }

    public void setClientId(String clientId) {
        this.clientId = clientId;
    }

    public String getCallId() {
        return callId;
    }

    public void setCallId(String callId) {
        this.callId = callId;
    }
}
